src/level.ts

```diff
@@ -1,151 +1,73 @@
-  } else if (myPrimestat() === $stat`Moxie`) {
-    tryEnsureEffect($effect`Pomp & Circumsands`);
-    ensureEffect($effect`Butt-Rock Hair`);
-    ensureEffect($effect`Superhuman Sarcasm`);
-    ensureEffect($effect`Cock of the Walk`);
-  }
-
-  if (myMp() < 200) {
-    eat(1, $item`magical sausage`);
-  }
-
-  // ML.
-  tryEnsureSong($skill`"Ur-Kel's Aria of Annoyance"`);
-  tryEnsureSkill($skill`Pride of the Puffin`);
-  tryEnsureSkill($skill`"Drescher's Annoying Noise"`);
-
-  // Combat.
-  tryEnsureSkill($skill`Carol of the Hells`);
-  ensureEffect($effect`Pisces in the Skyces`);
-
-  // Misc.
-  tryEnsureSong($skill`The Polka of Plenty`);
-  tryEnsureSong($skill`"Singer's Faithful Ocelot"`);
-  tryEnsureSkill($skill`Blood Bond`);
-  tryEnsureSkill($skill`Empathy of the Newt`);
-  tryEnsureSkill($skill`Leash of Linguini`);
-  tryEnsureSkill($skill`Carol of the Thrills`);
-  tryEnsureSkill($skill`Elemental Saucesphere`);
-  tryEnsureSkill($skill`Astral Shell`);
 }
 
-export function level(useResources = true) {
+export function level(useResources = true): void {
   if (myLevel() >= 13) return;
 
   // Put on some basic gear.
-  maximizeCached('mp');
-  if (myMp() < 200 && availableAmount($item`magical sausage`) + availableAmount($item`magical sausage casing`) > 0) {
-    eat(1, $item`magical sausage`);
-  }
-
-  // Start buffing. XP buffs first.
-  if (myPrimestat() === $stat`Muscle`) {
-    ensureEffect($effect`Muscle Unbound`);
-    ensureEffect($effect`Purpose`);
-    trySynthesize($effect`Synthesis: Movement`);
-  } else if (myPrimestat() === $stat`Mysticality`) {
-    ensureEffect($effect`Thaumodynamic`);
-    ensureEffect($effect`Category`);
-    trySynthesize($effect`Synthesis: Learning`);
-  } else if (myPrimestat() === $stat`Moxie`) {
-    ensureEffect($effect`So Fresh and So Clean`);
-    ensureEffect($effect`Perception`);
-    trySynthesize($effect`Synthesis: Style`);
-  }
+  maximizeCached(["MP"]);
 
-  // Campsite
-  if (haveEffect($effect`That's Just Cloud-Talk, Man`) === 0) {
-    visitUrl('place.php?whichplace=campaway&action=campaway_sky');
-  }
+  moodLevel().execute();
 
   // Daycare
-  if (getPropertyInt('_daycareGymScavenges') === 0) {
+  if (get("_daycareGymScavenges") === 0) {
     // Free scavenge.
-    visitUrl('choice.php?whichchoice=1336&option=2');
+    visitUrl("choice.php?whichchoice=1336&option=2");
   }
 
   // Bastille first.
-  if (getPropertyInt('_bastilleGames') === 0) {
-    if (availableAmount($item`Bastille Battalion control rig`) === 0) {
+  if (get("_bastilleGames") === 0) {
+    if (have($item`Bastille Battalion control rig`)) {
       use(1, $item`Bastille Battalion control rig loaner voucher`);
     }
-    cliExecute(`bastille ${myPrimestat() === $stat`Mysticality` ? 'myst' : myPrimestat()}`);
+    cliExecute(`bastille ${myPrimestat() === $stat`Mysticality` ? "myst" : myPrimestat()}`);
   }
 
   // Chateau rests.
-  if (getPropertyBoolean('chateauAvailable')) {
+  if (get("chateauAvailable")) {
```

Comment:
Consider having this be part of a gash-hop script, using the `prepareAscension` function